synapse/handlers/federation_event.py

```diff
@@ -2093,6 +2093,10 @@ async def _notify_persisted_event(
             event, event_pos, max_stream_token, extra_users=extra_users
         )
 
+        if event.type == EventTypes.Member and event.membership == Membership.JOIN:
+            # TODO retrieve the previous state, and exclude join -> join transitions
+            self._notifier.notify_user_joined_room(event.event_id, event.room_id)
+
```

Comment:
relevant comment: https://github.com/matrix-org/synapse/pull/13169#discussion_r918292716

Reply:
Ah yeah, looks like for whatever reason federation event hits the storage layer directly to persist events, rather than going through the code in `MessageHandler`